### PR TITLE
bookreader tests specific for archive.org usage

### PR DESCRIPTION
**Pull request by iisa**
### BR Tests & GHA Update:
- BR Test
  - move ia tests into repo from bookreader
  - add search helper that does not mock search request
- GHA update
  - Split up entry points so grokking gui is easier.  Jobs start when:
    -  any PR
    - merges to main
    - pushes to main
    - cron
 

- ✅ Tests work https://github.com/internetarchive/archiveorg-e2e-tests/actions/runs/3600461831/jobs/6065136616

### CODE REVIEW

File: .github/workflows/cron.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: PRs, Pushes, Cron
+
+on:
+  schedule:
+    - cron: "0 8 * * *"
```
Comment (dualcnhq) [inline .github/workflows/cron.yml:5]: q: is this workflow going to run every 8am or just once at 8am? why add this workflow?

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: Pull Request
```
Comment (dualcnhq) [inline .github/workflows/pull-request.yml:1]: 👍

File: .github/workflows/cron.yml (hunk 1 above)
Comment (iisa) [reply .github/workflows/cron.yml:5]: every. it has been here. just isolating it. plus we want something running at least once a day for now...

File: .github/workflows/books.yml [hunk 3]
```diff
@@ -8,22 +8,17 @@ jobs:
...
+      - name: Look inside current dir
+        run: ls -lh .
+      - name: Grab BR's test files
+        run: npm run port-br-files
       - name: 📚 📖 Books tests! 📖 📚
         uses: DevExpress/testcafe-action@latest
         with:
-          skip-install: true
-          args: "--config-file .testcaferc.js test/books"
+          args: "--color -q -c 2 chrome:headless,firefox:headless test/books"
```
Comment (bfalling) [inline .github/workflows/books.yml]: QUESTION: Just checking. Is it desired to hard-code the browsers being tested here?

File: package.json [hunk 4]
```diff
@@ -31,7 +32,7 @@
     "testcafe-reporter-slack": "^0.0.6"
   },
   "devDependencies": {
-    "@internetarchive/bookreader": "^5.0.0-38",
+    "@internetarchive/bookreader": "^5.0.0-49",
```
Comment (bfalling) [inline package.json:35]: QUESTION: Is it desired to have hard-coded versions here? Does that mean that we'll have to update version numbers in the e2e repo whenever we change package versions? Would it be better to always test the latest instead, so these versions don't need to be constantly updated? Or maybe I'm not understanding...

File: test/books/details-public-book.test.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import { runSearchTests } from './helpers/search-inside.js';
+
+const localURL = 'https://archive.org/details/';
+const books = [
+  'birdbookillustra00reedrich', // publicDomain
+  // 'pianoservicingtu00rebl_0', // borrowable,
+  // 'adventuresoftoms00twaiiala', // has chapter markers
+  // 'gendaitankashu00meijuoft', // Right to Left book
+  // 'gov.uspto.patents.application.10074026', // multiple files,
+  // 'theworksofplato01platiala', //bookreader demo
```
Comment (bfalling) [inline test/books/details-public-book.test.js:15]: QUESTION: How are these commented-out lines used? If we are to keep commented out code, maybe we should explain why it's there and how to use it...?

**Review by bfalling — APPROVED** [review]
A buncha questions. No blockers found.

File: package.json (hunk 4 above)
Comment (iisa) [reply package.json]: i will look into pointing to `latest` tag...

File: test/books/details-public-book.test.js (hunk 5 above)
Comment (iisa) [reply test/books/details-public-book.test.js:16]: these books are not yet supported with their specific tests. good to keep them there for specific examples. i've added a comment.

**Pull request merged**